### PR TITLE
add missing $ in docs

### PR DESCRIPTION
**Pull request by bernt-matthias**

### CODE REVIEW

**Review by mvdbeek — APPROVED** [review]
That's clearer, thanks. Just curious, are you sure this doesn't work ?
Once you're inside ${} you should be in a python scope (that's why you can call out to `os.path` for instance), and that scope should include parameter names.

**Comment by bernt-matthias** [comment]
> Just curious, are you sure this doesn't work ?

Just have seen this failing. Also wondering how/why all these singe quotes within single quotes are working. Maybe this is another problem.

**Comment by mvdbeek** [comment]
Thanks, if you drop the commit I'm happy to merge

**Comment by bernt-matthias** [comment]
done

File: doc/source/dev/data_types.md [hunk 1]
```diff
@@ -404,8 +404,8 @@ class BasicComposite(Text):
 These files can be specified on the command line in the following fashion:
 
 ```xml
-<command>someTool.sh '$input1' '${os.path.join(input1.extra_files_path, 'results.txt')}'
-'${os.path.join(input1.extra_files_path, 'results.dat')}' '$output1'</command>
+<command>someTool.sh '$input1' '${os.path.join($input1.extra_files_path, 'results.txt')}'
```
Comment (dannon) [inline doc/source/dev/data_types.md:407]: Are you positive you need `$input1` here?  I was also almost certain that once in the python context (the outer `${....}`, you wouldn't need this.

Comment (bernt-matthias) [reply doc/source/dev/data_types.md:407]: Not 100% sure .. so I re-added the test that should fail (this time with out the typo .. I forgot the closing `}` the first time, ie in https://github.com/galaxyproject/galaxy/commit/f065a38b2b2ae434b2166e2f84f08ecb0883e080).

Comment (bernt-matthias) [reply doc/source/dev/data_types.md:407]: So the framework tests are failing as expected. If ran in planemo the full Traceback is:

```
Traceback (most recent call last):
  File "/tmp/tmpkifhuuke/galaxy-dev/lib/galaxy/util/__init__.py", line 1053, in unicodify
    value = text_type(value)
  File "/home/berntm/.planemo/gx_venv_3.7_release_20.05/lib/python3.6/site-packages/Cheetah/Template.py", line 1053, in __unicode__
    return getattr(self, mainMethName)()
  File "cheetah_DynamicallyCompiledCheetahTemplate_1596627208_607981_46507.py", line 89, in respond
AttributeError: 'builtin_function_or_method' object has no attribute 'extra_files_path'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/tmpkifhuuke/galaxy-dev/lib/galaxy/util/__init__.py", line 1055, in unicodify
    value = str(value)
  File "/home/berntm/.planemo/gx_venv_3.7_release_20.05/lib/python3.6/site-packages/Cheetah/Template.py", line 1053, in __unicode__
    return getattr(self, mainMethName)()
  File "cheetah_DynamicallyCompiledCheetahTemplate_1596627208_607981_46507.py", line 89, in respond
AttributeError: 'builtin_function_or_method' object has no attribute 'extra_files_path'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/tmpkifhuuke/galaxy-dev/lib/galaxy/util/template.py", line 67, in fill_template
    return unicodify(t)
  File "/tmp/tmpkifhuuke/galaxy-dev/lib/galaxy/util/__init__.py", line 1061, in unicodify
    msg = "Value '%s' could not be coerced to Unicode: %s('%s')" % (value, type(e).__name__, e)
  File "/home/berntm/.planemo/gx_venv_3.7_release_20.05/lib/python3.6/site-packages/Cheetah/Template.py", line 1053, in __unicode__
    return getattr(self, mainMethName)()
  File "cheetah_DynamicallyCompiledCheetahTemplate_1596627208_607981_46507.py", line 89, in respond
AttributeError: 'builtin_function_or_method' object has no attribute 'extra_files_path'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/tmpkifhuuke/galaxy-dev/lib/galaxy/jobs/runners/__init__.py", line 236, in prepare_job
    job_wrapper.prepare()
  File "/tmp/tmpkifhuuke/galaxy-dev/lib/galaxy/jobs/__init__.py", line 1107, in prepare
    self.command_line, self.extra_filenames, self.environment_variables = tool_evaluator.build()
  File "/tmp/tmpkifhuuke/galaxy-dev/lib/galaxy/tools/evaluation.py", line 475, in build
    raise e
  File "/tmp/tmpkifhuuke/galaxy-dev/lib/galaxy/tools/evaluation.py", line 471, in build
    self.__build_command_line()
  File "/tmp/tmpkifhuuke/galaxy-dev/lib/galaxy/tools/evaluation.py", line 496, in __build_command_line
    command_line = fill_template(command, context=param_dict, python_template_version=self.tool.python_template_version)
  File "/tmp/tmpkifhuuke/galaxy-dev/lib/galaxy/util/template.py", line 111, in fill_template
    python_template_version=python_template_version,
  File "/tmp/tmpkifhuuke/galaxy-dev/lib/galaxy/util/template.py", line 113, in fill_template
    raise first_exception or e
  File "/tmp/tmpkifhuuke/galaxy-dev/lib/galaxy/util/template.py", line 67, in fill_template
    return unicodify(t)
  File "/tmp/tmpkifhuuke/galaxy-dev/lib/galaxy/util/__init__.py", line 1061, in unicodify
    msg = "Value '%s' could not be coerced to Unicode: %s('%s')" % (value, type(e).__name__, e)
  File "/home/berntm/.planemo/gx_venv_3.7_release_20.05/lib/python3.6/site-packages/Cheetah/Template.py", line 1053, in __unicode__
    return getattr(self, mainMethName)()
  File "cheetah_DynamicallyCompiledCheetahTemplate_1596627208_5605676_14480.py", line 87, in respond
AttributeError: 'builtin_function_or_method' object has no attribute 'extra_files_path'
FAIL
```

Comment (mvdbeek) [reply doc/source/dev/data_types.md:407]: Are you sure that the tool you tested used `input1`, not `input` (and then `input` might be the python builtin)? But okay, that should be reason enough to use the `$` variant.

Comment (bernt-matthias) [reply doc/source/dev/data_types.md:407]: You are right, input seems to be a builtin. If I change the tool input name to something else (eg `blahfasel`) I get 

```
Traceback (most recent call last):
  File "/tmp/tmpb5jlyh30/galaxy-dev/lib/galaxy/util/__init__.py", line 1053, in unicodify
    value = text_type(value)
  File "/home/berntm/.planemo/gx_venv_3.7_release_20.05/lib/python3.6/site-packages/Cheetah/Template.py", line 1053, in __unicode__
    return getattr(self, mainMethName)()
  File "cheetah_DynamicallyCompiledCheetahTemplate_1596628047_2029321_19077.py", line 89, in respond
NameError: name 'blahfasel' is not defined

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/tmpb5jlyh30/galaxy-dev/lib/galaxy/util/__init__.py", line 1055, in unicodify
    value = str(value)
  File "/home/berntm/.planemo/gx_venv_3.7_release_20.05/lib/python3.6/site-packages/Cheetah/Template.py", line 1053, in __unicode__
    return getattr(self, mainMethName)()
  File "cheetah_DynamicallyCompiledCheetahTemplate_1596628047_2029321_19077.py", line 89, in respond
NameError: name 'blahfasel' is not defined

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/tmpb5jlyh30/galaxy-dev/lib/galaxy/util/template.py", line 67, in fill_template
    return unicodify(t)
  File "/tmp/tmpb5jlyh30/galaxy-dev/lib/galaxy/util/__init__.py", line 1061, in unicodify
    msg = "Value '%s' could not be coerced to Unicode: %s('%s')" % (value, type(e).__name__, e)
  File "/home/berntm/.planemo/gx_venv_3.7_release_20.05/lib/python3.6/site-packages/Cheetah/Template.py", line 1053, in __unicode__
    return getattr(self, mainMethName)()
  File "cheetah_DynamicallyCompiledCheetahTemplate_1596628047_2029321_19077.py", line 89, in respond
NameError: name 'blahfasel' is not defined

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/tmpb5jlyh30/galaxy-dev/lib/galaxy/jobs/runners/__init__.py", line 236, in prepare_job
    job_wrapper.prepare()
  File "/tmp/tmpb5jlyh30/galaxy-dev/lib/galaxy/jobs/__init__.py", line 1107, in prepare
    self.command_line, self.extra_filenames, self.environment_variables = tool_evaluator.build()
  File "/tmp/tmpb5jlyh30/galaxy-dev/lib/galaxy/tools/evaluation.py", line 475, in build
    raise e
  File "/tmp/tmpb5jlyh30/galaxy-dev/lib/galaxy/tools/evaluation.py", line 471, in build
    self.__build_command_line()
  File "/tmp/tmpb5jlyh30/galaxy-dev/lib/galaxy/tools/evaluation.py", line 496, in __build_command_line
    command_line = fill_template(command, context=param_dict, python_template_version=self.tool.python_template_version)
  File "/tmp/tmpb5jlyh30/galaxy-dev/lib/galaxy/util/template.py", line 111, in fill_template
    python_template_version=python_template_version,
  File "/tmp/tmpb5jlyh30/galaxy-dev/lib/galaxy/util/template.py", line 113, in fill_template
    raise first_exception or e
  File "/tmp/tmpb5jlyh30/galaxy-dev/lib/galaxy/util/template.py", line 67, in fill_template
    return unicodify(t)
  File "/tmp/tmpb5jlyh30/galaxy-dev/lib/galaxy/util/__init__.py", line 1061, in unicodify
    msg = "Value '%s' could not be coerced to Unicode: %s('%s')" % (value, type(e).__name__, e)
  File "/home/berntm/.planemo/gx_venv_3.7_release_20.05/lib/python3.6/site-packages/Cheetah/Template.py", line 1053, in __unicode__
    return getattr(self, mainMethName)()
  File "cheetah_DynamicallyCompiledCheetahTemplate_1596628047_1631348_56014.py", line 87, in respond
NameError: name 'blahfasel' is not defined
FAIL
```

Comment (bernt-matthias) [reply doc/source/dev/data_types.md:407]: With the extra `$` it works.

Comment (dannon) [reply doc/source/dev/data_types.md:407]: Yeah, sure enough, just tried it locally, too.  This definitely did not work like I expected it to.

**Review by dannon — APPROVED** [review]

**Comment by dannon** [comment]
Annnd, I just saw the failing tool test you added on a page refresh here after the merge.  I'll fix it.

**Comment by bernt-matthias** [comment]
> Annnd, I just saw the failing tool test you added on a page refresh here after the merge. I'll fix it.

cool

**Pull request merged**